### PR TITLE
chore: Fix bug and provide context to descriptor 404s

### PR DESCRIPTION
**Pull request by Attamusc**

### CODE REVIEW

**Comment by Aintaer** [comment]
:+1:

**Comment by joesepi** [comment]
:+1:

File: test/specs/client.js [hunk 1]
```diff
@@ -257,6 +257,35 @@ define(['chai-as-promised', 'lib/client', 'lib/resource'], function(chaiAsPromis
...
+          foo: [404, { "Content-Type": "text/plain" }, ''],
+        };
+
+        it('returns an "Unable to get descriptor" error', function() {
+          var client = new Client('/v1');
+
+          server.respondWith('OPTIONS', '/v1', responses.root);
+          server.respondWith('OPTIONS', '/v1/foo', responses.foo);
+
+          return expect(client.walk('foo')).to.be.rejectedWith(new RegExp('Unable to get descriptor for uri "/v1/foo"'));
```
Comment (mikesherov) [inline test/specs/client.js]: use a regex literal here?

**Comment by mikesherov** [comment]
:+1:

File: test/specs/client.js (hunk 1 above)
Comment (Attamusc) [inline test/specs/client.js]: jscs got really sad about that for some reason

**Pull request merged**